### PR TITLE
Add a health check API to serial-vault

### PR DESCRIPTION
**Pull request by ejfinneran**
`/v1/health` checks to ensure that the database is accessible using the
db.Ping() function.

### CODE REVIEW

File: service/handlers.go [hunk 1]
```diff
@@ -84,6 +89,23 @@ func VersionHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// HealthHander is the API method to return if the app is up and db.Ping() doesn't return an error
+func HealthHander(w http.ResponseWriter, r *http.Request) {
```
Comment (slimjim777) [inline service/handlers.go]: Minor comment: Hander should be Handler
However, we are in the process of a refactor and removing the team Handler altogether from these functions.

File: service/handlers_test.go [hunk 2]
```diff
@@ -643,6 +643,45 @@ func TestVersionHandler(t *testing.T) {
...
+
+}
+
+func TestHealthHandlerUnhealthy(t *testing.T) {
+
+	config := config.Settings{KeyStoreType: "filesystem", KeyStorePath: "../keystore", JwtSecret: "SomeTestSecretValue"}
+	datastore.Environ = &datastore.Env{DB: &datastore.ErrorMockDB{}, Config: config}
+	response, _ := sendRequestHealth(t, "GET", "/v1/health", nil)
+
+	if response.Code != 500 {
```
Comment (slimjim777) [inline service/handlers_test.go]: This test is failing as the method always returns an HTTP 200 response. You'll need to set the response in the HealthCheck method. Also, we prefer HTTP 400 errors for understood application errors - that way we know that if we get an error 500 we know that something unexpected has happened.

So, for example, the health check may actually fail when the service is first deployed as the postgresql unit takes longer to start up and initialize than the SV units.

**Comment by ejfinneran** [comment]
I fixed the tests and function names but it looks like `db.Ping()` doesn't handle connections dying very well so it's sending back false positives right now.

Need to look at better ways check DB health before merging this.

**Comment by ejfinneran** [comment]
I opted for running a simple query (`select 1`) . I tested the endpoint while terminating the database container and the endpoint started return 400 as expected.

**Pull request merged**